app/models/course/assessment/answer/programming_file.rb

```diff
@@ -10,6 +10,22 @@ class Course::Assessment::Answer::ProgrammingFile < ActiveRecord::Base
   has_many :annotations, class_name: Course::Assessment::Answer::ProgrammingFileAnnotation.name,
                          dependent: :destroy, foreign_key: :file_id, inverse_of: :file
 
+  # Returns the code from start line to end line.
+  #
+  # @param [Integer] line_start 1 based line number (first line is 1).
+  # @param [Integer] line_end 1 based line number (first line is 1).
+  # @return [String|nil] the code in the lines specified. `nil` will be returned if the input is not
+  #   valid.
+  def code_at_lines(line_start, line_end)
```

Comment:
I am still for it being a view method. Otherwise, use a view helper.

The model should handle the clamping of the values to valid ranges, but the view helper should handle the 1 to 0 change.